tools/ci/docker/linux/Dockerfile

```diff
@@ -84,8 +85,10 @@ WORKDIR /tools
 FROM nuttx-toolchain-base AS nuttx-toolchain-arm
 # Download the latest ARM GCC toolchain prebuilt by ARM
 RUN mkdir gcc-arm-none-eabi && \
-  curl -s -L  "https://developer.arm.com/-/media/Files/downloads/gnu-rm/10.3-2021.10/gcc-arm-none-eabi-10.3-2021.10-x86_64-linux.tar.bz2" \
-  | tar -C gcc-arm-none-eabi --strip-components 1 -xj
+  curl -s -L  "https://developer.arm.com/-/media/Files/downloads/gnu/11.3.rel1/binrel/arm-gnu-toolchain-11.3.rel1-x86_64-arm-none-eabi.tar.xz" \
+  | tar -C gcc-arm-none-eabi --strip-components 1 -xJ  \
+  && curl -s -L -O "https://raw.githubusercontent.com/apache/incubator-nuttx/master/tools/ci/patch/arm-none-eabi-workaround-for-newlib-version-break.patch" \
```

Comment:
This should be copied in instead of downloaded so that it matches the file in the repo at build time.  I would recommend adding a step to copy the patches to be at the same level as the dockerfile. The copy in the dockerfile cannot traverse up the tree.

Reply:
It seems all toolchains are downloaded dynamically, do we need to do a similar optimization for all toolchains? This will save docker build time

Reply:
@btashton could you explain more